apps/src/code-studio/assets/SpriteUpload.js

```diff
@@ -22,20 +56,30 @@ export default class SpriteUpload extends React.Component {
         <h1>Sprite Upload</h1>
         <form onSubmit={this.handleSubmit}>
           <label>
-            <h4> Sprite Category: </h4>
-            <input type="text" />
+            <h4>Sprite Category:</h4>
+            <input
+              type="text"
+              value={this.state.category}
+              onChange={this.handleCategoryChange}
+            />
           </label>
           <label>
-            <h4> Select Sprite to Add to Library: </h4>
-            <input type="file" ref="uploader" onChange={this.handleChange} />
+            <h4>Select Sprite to Add to Library:</h4>
+            <input
+              type="file"
+              accept="image/png"
+              ref="uploader"
+              onChange={this.handleImageChange}
+            />
           </label>
           <br />
           <label>
-            <h4> Image Preview: </h4>
-            <img src={this.state.file} />
+            <h4>Image Preview:</h4>
+            <img src={this.state.filePreviewURL} />
           </label>
           <br />
           <button type="submit">Upload to Library</button>
+          <p id="sprite-upload-error">{this.state.uploadState}</p>
```

Comment:
two things:

1. we don't need this id anymore
2. what does this end up looking like? it might be easy to miss this message if we don't add styling to distinguish between success/error here.

Reply:
It is pretty subtle. Here are two screenshots: 
![Screenshot from 2021-04-28 14-49-10](https://user-images.githubusercontent.com/2959170/116484741-966b8680-a83e-11eb-8a28-fccfc113a4bb.png)
![Screenshot from 2021-04-28 11-22-22](https://user-images.githubusercontent.com/2959170/116484746-99667700-a83e-11eb-869a-f40a79206248.png)

Reply:
I'm happy to add styling to make it a little bigger/redder if it fails, but I also want to get feedback from the curriculum writers on what their preferred behavior is here before diving too deep - an alert message or maybe removing the file preview, might also suit.

Reply:
I just saw your comment below and think adding some quick styling here for now and getting curriculum writer feedback on the ideal error behavior in the future is a good plan for now.

Reply:
Now this looks like:
![Screenshot from 2021-04-28 17-02-12](https://user-images.githubusercontent.com/2959170/116487608-388e6d00-a845-11eb-950a-1b259d797985.png)
![Screenshot from 2021-04-28 17-01-34](https://user-images.githubusercontent.com/2959170/116487614-3c21f400-a845-11eb-9eea-18d2d5577014.png)